Rewrite the 'PivotGrid - Chart Integration' demo description (#1329)
DevExpress/devextreme-demos: JSDemos/Demos/PivotGrid/ChartIntegration/Angular/description.md

> PivotGrid allows you to integrate the [Chart](/Documentation/ApiReference/UI_Components/dxChart/) component to visualize data. The Chart is synchronized with the PivotGrid and visualizes currently displayed total summary values. Expand a row or column header in the PivotGrid below to see how it works.

Reviewer: "currently" is a strong word that implies grammatical definiteness.

```suggestion
PivotGrid allows you to integrate the [Chart](/Documentation/ApiReference/UI_Components/dxChart/) component to visualize data. The Chart is synchronized with the PivotGrid and visualizes the currently displayed total summary values. Expand a row or column header in the PivotGrid below to see how it works.
```

## Bind the Chart to the PivotGrid

> To bind these two components, create a Chart and pass its instance as the first argument to the [bindChart(chart, integrationOptions)](/Documentation/ApiReference/UI_Components/dxPivotGrid/Methods/#bindChartchart_integrationOptions) method. You can call this method at any point of your application. In this demo, we call this method in the <a href="https://angular.io/api/core/AfterViewInit" target="_blank">ngAfterViewInit</a> lifecycle hook and use instances to bind the components.

Reviewer: > To bind these two components, create a Chart and pass **its instance** as the first argument to the [bindChart(chart, integrationOptions)](/Documentation/ApiReference/UI_Components/dxPivotGrid/Methods/#bindChartchart_integrationOptions) method.

We can easily simplify this. You already said "create a Chart", and this created chart is already a Chart instance.

To bind these two components, create a Chart and pass it as the first argument to the [bindChart(chart, intergrationOptions)](/Documentation/ApiReference/UI_Components/dxPivotGrid/Methods/#bindChartchart_integrationOptions) method.

> You can call this method **at** any point **of** your application.

"At" is a correct preposition because it answers the question "Where, at which position". I do not believe that "of" is a correct preposition because it speaks of possession. At any point that belongs to your application. However, you actually meant that you can call the mentioned method **at** any point **in** code of your application (that is, "where" - at any point and "where" - in code, in your application). Therefore, we'll get this version: "You can call this method at any point in your application". Moreover, the following sentence supports this idea.

> In this demo, we call this method in the <a href="https://angular.io/api/core/AfterViewInit" target="_blank">ngAfterViewInit</a> lifecycle hook and **use instances to bind the components**.

To my regret, I failed to understand this part despite my best efforts. Whose instances do you use? Aren't "created components" and "component instances" the same? You instantiate a class and get a "class instance" or just "object", which is the actual component in your working application. In my opinion, we can just omit this part and nothing will change.

> If you use only the first argument and omit **integrationOptions**, default settings are used: row field values form chart [series](/Documentation/ApiReference/UI_Components/dxChart/Configuration/series/), and column field values form [arguments](/Documentation/ApiReference/UI_Components/dxChart/Configuration/argumentAxis/). The Chart displays values of each data field on a separate [value axis](/Documentation/ApiReference/UI_Components/dxChart/Configuration/valueAxis/). If you need to customize the resulting chart, specify the **integrationOptions** argument.  

Reviewer: I believe we can optimize the first sentence a bit.

```suggestion
If you specify only the first argument and omit **integrationOptions**, default settings are in effect: row and column field values constitute chart [series](/Documentation/ApiReference/UI_Components/dxChart/Configuration/series/) and [arguments](/Documentation/ApiReference/UI_Components/dxChart/Configuration/argumentAxis/), respectively. The Chart displays values of each data field on a separate [value axis](/Documentation/ApiReference/UI_Components/dxChart/Configuration/valueAxis/). If you need to customize the resulting chart, specify the **integrationOptions** argument.  
```

## Convert PivotGrid Fields to Chart Series
You can use the following **integrationOptions** to specify how PivotGrid fields should be converted to chart series:

- **alternateDataFields**: Boolean       
Specifies whether to alternate data fields in the resulting chart. This demo sets the **alternateDataFields** field to **false**.

- **dataFieldsDisplayMode**: *"singleAxis"* | *"splitAxes"* | *"splitPanes"*     

> Accepts one of the following values.   

Reviewer: This line introduces a list, shouldn't such a line end with a colon?

```suggestion
Accepts one of the following values:
```

>     - *"singleAxis"* - values of all data fields are displayed on a single [value axis](/Documentation/ApiReference/UI_Components/dxChart/Configuration/valueAxis/).  

Reviewer: Can we say?

```suggestion
    - *"singleAxis"* - All data field values are displayed on a single [value axis](/Documentation/ApiReference/UI_Components/dxChart/Configuration/valueAxis/).  
```

>     - *"splitAxes"* - each data field gets an individual value axis. Applies only if **putDataFieldsInto** is *"series"*.  

Reviewer: Why does it "gets"?

```suggestion
    - *"splitAxes"* - each data field has an individual value axis. Applies only if **putDataFieldsInto** is *"series"*.
```

>     - *"splitPanes"* - each data field gets an individual [pane](/Documentation/ApiReference/UI_Components/dxChart/Configuration/panes/). Applies only if **putDataFieldsInto** is *"series"*.    

Reviewer: ```suggestion
    - *"splitPanes"* - each data field has an individual [pane](/Documentation/ApiReference/UI_Components/dxChart/Configuration/panes/). Applies only if **putDataFieldsInto** is *"series"*.
```

>     Applies only if the PivotGrid contains several [data fields](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/). In this demo, the **dataFieldsDisplayMode** property's value is *"splitPanes"*.

Reviewer: Does this line still speak about the **dataFieldsDisplayMode**? And again, is it a field or property? The initial line speaks about all listed items as "fields". As for this line, I would have moved this information up, before the value list. Right now this limitation does not look as a part of the **dataFieldsDisplayMode** description.

Author: My bad, it is a field. But this limitation relates to the **splitPanes** field that depends on the **dataFieldsDisplayMode** field's value.

Reviewer: Wait. I thought that **splitPanes** is one of the possible values of the **dataFieldsDisplayMode** field, and now you state that **splitPlanes** is also a field. I'm getting more and more confused, actually.

>     Applies only if the PivotGrid contains several [data fields](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/). In this demo, the **dataFieldsDisplayMode** field's value is *"splitPanes"*.

Reviewer 2: Is this description missing the option name?

- **inverted**: Boolean      

> If **true**, [row field](/Documentation/Guide/UI_Components/PivotGrid/Visual_Elements/#Headers) values go to the [chart arguments](/Documentation/ApiReference/UI_Components/dxChart/Configuration/argumentAxis/) and [column field](/Documentation/Guide/UI_Components/PivotGrid/Visual_Elements/#Headers) values form [series](/Documentation/ApiReference/UI_Components/dxChart/Configuration/series/). If **false** - vice versa.

Reviewer: Looks like the **false** value corresponds to the "normal" order of things, and **true** inverts the order of sources. I would have explained exactly that, starting with **false**. I believe that "invert" or "inverted" should be in your explanation.

- **putDataFieldsInto**: *"series"* | *"args"*     

> Accepts one of the following values.

Reviewer: ```suggestion
Accepts one of the following values:
```

>     - *"args"* - data fields go to the Chart arguments.  

Reviewer: What do you mean by the verb "go" exactly?

Author: I think that most precise word is 'constitute' and I should use this word instead.

Reviewer: Yeah, I agree, the verb "to constitute" would be the most appropriate.

    - *"series"* - data fields constitute the Chart series.

## Advanced Customization

> The **integrationOptions** object also can contain the following functions that allow you to process PivotGrid data before visualization and customize the Chart and its series:

Reviewer 2: ```suggestion
The **integrationOptions** object can also contain the following functions that allow you to process PivotGrid data before visualization, and customize the Chart and its series:
```

- **processCell**       

> Processes a PivotGrid cell's data before the Chart displays it.

Reviewer: How one can "process" a cell's data? Of which actions do you speak? Like "apply a formatting pattern" or something else that I can't even imagine? The verb "to process" is very general by its nature, it could mean almost anything and conveys nothing in particular at the same time.

Author: That's the point. processCell allows customers to do almost anything with a cell's data.

Reviewer: Alright, can it modify values or the "process" is limited to mostly cosmetic changes? As for me, I'd like to see a couple of examples in this line.

Author: There are two main problems with this suggestion that I discussed with @RomanTsukanov .
- This description is already huge and examples will definetely overload it.
- The more important fact is that this function allows developers to modify data in any way they want (not only cosmetic changes are possible). There aren't any particular examples since this function is designed to give developers absolute freedom.

Reviewer: This means you all set then. I will not insist.

- **customizeChart**        
Allows you to customize the Chart's appearance.

- **customizeSeries**       

> Allows you to customize the Chart's series.

Reviewer: I have no questions about "customizeChart" because you detailed that it's all about appearance. And what do you mean by series customization? An example would be great.

Author: Once again, there are multiple settings related to series. These setting are different for each series type -  'area', 'bar', 'bubble', 'candlestick' , etc. But series are, in fact, part of Chart's appearance.

Reviewer: I see. This is basically a subset of Chart appearance settings. I would have tried to clarify this fact, maybe a couple of examples would be in order.

Author: The same point as above - there is almost infinite number of ways to modify the resulting chart. It's up to developer to decide what to do.

Reviewer: OK, I see. The matter is closed then.

> Refer to the **bindChart(chart, integrationOptions)** method's [description](/Documentation/ApiReference/UI_Components/dxPivotGrid/Methods/#bindChartchart_integrationOptions) for more information on these functions. 

Reviewer: You can place a link directly on the method's name to save space and simplify your sentence:

```suggestion
Refer to the [bindChart(chart, integrationOptions)](/Documentation/ApiReference/UI_Components/dxPivotGrid/Methods/#bindChartchart_integrationOptions) method description for more information on these functions.
```
